js/api-docs-redirect.js

```diff
@@ -1,7 +1,15 @@
 ;(function redirectApiDocsHash() {
-  if (window.location.pathname.match(/^\/developers\/api\/?$/) && window.location.hash) {
-    switch(window.location.hash.toLowerCase()) {
-      case '#introduction': window.location.assign('/developers/api')
+  function getRedirectRoute(hash) {
+    switch(hash) {
+      case '#introduction': return '/developers/api'
+      case '#authentication': return '/developers/api/authentication'
+      case '#api-keys':
+      case '#key-restrictions': return `/developers/api/authentication${hash}`
+      default: return `/developers/api${hash}`
     }
   }
+
+  if (window.location.pathname.match(/^\/developers\/api\/?$/) && window.location.hash) {
```

Comment:
We should add one more check to this conditional to prevent a redirect if `getRedirectRoute` returns the current path/hash. If I visit `/developers/api#not-found-anywhere`, the code in the if statement will run, and `getRedirectRoute` will return `/developers/api#not-found-anywhere`, causing an endless redirect to the current page.

Reply:
Ah, thanks for catching that. Updating!